### DISCUSSION TITLE
Import from url / endpoint / webthings

### DISCUSSION THREAD

**Discussion by jonnyhotchkiss** [brian3kb/digichain#19] [Ideas]
Is it possible to add web import options, to import from "open" and/or "authenticated" sources like

YouTube
FreeSound
SoundCloud
Spotify
_ad infinitum_

& Is this something you've considered, or reviewed?

(Still/Currently hooked on ideas like of getting a universal audio clipboard to work, so the copy/paste functionality in apps is extended, and/or the user-workflow is optimised, even user workarounds like (behind the scenes) write_to_filesystem, if 'AudioBuffers' aren't readily available....

Thanks in advance
Jonny (currently Windows 10 Home, SonicPi Student with a head full of ideas)

**Comment by brian3kb** [comment]
Apologies for the extremely late reply to this! If the source support open cross origin resource sharing, then the linked audio can be imported by dragging the link into DigiChain, the only source of a large number of audio files that I know of that works well is the Library of Congress Citizen DJ library https://citizen-dj.labs.loc.gov/

Choosing a sound and dragging the 'Download' button into DigiChain will load the sample directly into the list - in theory, this will work for any site that allows the link to be opened by a different domains origin - unfortunately all those listed in the OP are locked down to the same origin, or are not friendly to third-party tools due to licensing (i.e. Spotify, YouTube, SoundCloud).
